packages/core-data/src/resolvers.js

```diff
@@ -50,7 +51,10 @@ export async function* getEntityRecord( state, kind, name, key ) {
 	if ( ! entity ) {
 		return;
 	}
-	const record = await apiRequest( { path: `${ entity.baseUrl }/${ key }?context=edit` } );
+
+	// Makes sure the media will be fetched even for contributors.
+	const context = 'media' === entity.name ? 'view' : 'edit';
```

Comment:
We really shouldn't have to manually assign these contexts, going to be difficult to maintain. I don't have a great solution right now; the whole context thing is causing difficulties in general.